core/src/main/java/com/linecorp/armeria/server/metric/PrometheusExpositionService.java

```diff
@@ -35,20 +41,45 @@
  * Exposes Prometheus metrics in <a href="https://prometheus.io/docs/instrumenting/exposition_formats/">text
  * format 0.0.4</a>.
  */
-public final class PrometheusExpositionService extends AbstractHttpService {
+public final class PrometheusExpositionService extends AbstractHttpService implements TransientHttpService {
 
     private static final MediaType CONTENT_TYPE_004 = MediaType.parse(TextFormat.CONTENT_TYPE_004);
 
+    /**
+     * Returns a new {@link PrometheusExpositionService} that exposes Prometheus metrics from the specified
+     * {@link CollectorRegistry}.
+     */
+    public static PrometheusExpositionService of(CollectorRegistry collectorRegistry) {
+        return new PrometheusExpositionService(collectorRegistry, defaultTransientServiceActions());
+    }
+
+    /**
+     * Returns a new {@link PrometheusExpositionServiceBuilder} created with the specified
+     * {@link CollectorRegistry}.
+     */
+    public static PrometheusExpositionServiceBuilder builder(CollectorRegistry collectorRegistry) {
+        return new PrometheusExpositionServiceBuilder(collectorRegistry);
+    }
+
     private final CollectorRegistry collectorRegistry;
+    private final Map<ActionType, Boolean> transientServiceActions;
 
     /**
      * Creates a new instance.
      *
      * @param collectorRegistry Prometheus registry
+     *
+     * @deprecated Use {@link #of(CollectorRegistry)}.
      */
+    @Deprecated
     public PrometheusExpositionService(CollectorRegistry collectorRegistry) {
-        requireNonNull(collectorRegistry, "collectorRegistry");
-        this.collectorRegistry = collectorRegistry;
+        this(collectorRegistry, defaultTransientServiceActions());
+    }
+
+    PrometheusExpositionService(CollectorRegistry collectorRegistry,
+                                EnumMap<ActionType, Boolean> transientServiceActions) {
+        this.collectorRegistry = requireNonNull(collectorRegistry, "collectorRegistry");
+        this.transientServiceActions = ImmutableMap.copyOf(transientServiceActions);
```

Comment:
Could use `Maps.immutableEnumMap()` to avoid addtional copies?
https://github.com/google/guava/blob/7ef9f97cad23efee4cc18c63b5f3c90ca320f131/guava/src/com/google/common/collect/Maps.java#L148-L152